src/app/WriteHandler.cpp

```diff
@@ -202,6 +202,80 @@ CHIP_ERROR WriteHandler::SendWriteResponse(System::PacketBufferTLVWriter && aMes
     return err;
 }
 
+void WriteHandler::DeliverListWriteBegin(const ConcreteAttributePath & aPath)
+{
+    if (auto * attrOverride = GetAttributeAccessOverride(aPath.mEndpointId, aPath.mClusterId))
+    {
+        attrOverride->OnListWriteBegin(aPath);
+    }
+}
+
+void WriteHandler::DeliverListWriteEnd(const ConcreteAttributePath & aPath, bool writeWasSuccessful)
+{
+    if (auto * attrOverride = GetAttributeAccessOverride(aPath.mEndpointId, aPath.mClusterId))
+    {
+        attrOverride->OnListWriteEnd(aPath, writeWasSuccessful);
+    }
+}
+
+void WriteHandler::DeliverFinalListWriteEnd(bool writeWasSuccessful)
+{
+    if (mProcessingAttributePath.HasValue() && mProcessingAttributeIsList)
+    {
+        DeliverListWriteEnd(mProcessingAttributePath.Value(), writeWasSuccessful);
+    }
+    mProcessingAttributePath.ClearValue();
+}
+
+CHIP_ERROR WriteHandler::DeliverFinalListWriteEndForGroupWrite(bool writeWasSuccessful)
+{
+    VerifyOrReturnError(mProcessingAttributePath.HasValue() && mProcessingAttributeIsList, CHIP_NO_ERROR);
+
+    Credentials::GroupDataProvider::GroupEndpoint mapping;
+    Credentials::GroupDataProvider * groupDataProvider = Credentials::GetGroupDataProvider();
+    Credentials::GroupDataProvider::EndpointIterator * iterator;
+
+    GroupId groupId         = mpExchangeCtx->GetSessionHandle()->AsIncomingGroupSession()->GetGroupId();
+    FabricIndex fabricIndex = GetAccessingFabricIndex();
+
+    iterator = groupDataProvider->IterateEndpoints(fabricIndex);
+    VerifyOrReturnError(iterator != nullptr, CHIP_ERROR_NO_MEMORY);
```

Comment:
Don't we need to still do `mProcessingAttributePath.ClearValue()` in this case?

I would recommend that after our initial "we have a list attribute" check we just copy `mProcessingAttributePath.Value()` into an on-stack ConcreteAttributePath, clear it, then do the work in this method.  That will guarantee that we clear `mProcessingAttributePath`, which we need to do, because otherwise we will end up in `DeliverFinalListWriteEnd` with a path that has a wildcard endpoint.

Reply:
Yes, we should, fixed.